EIPS/eip-2677.md

```diff
@@ -0,0 +1,76 @@
+---
+eip: 2677
+title: Limit size of `initcode`
+author: Martin Holst Swende (@holiman), Pawel Bylica (@chfast), Alex Beregszazi (@axic)
+discussions-to: https://ethereum-magicians.org/t/eip-2677-limit-size-of-initcode/4550
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-05-18
+---
+
+## Simple Summary
+
+Enforce a maximum size limit (`max_initcode_size`) of `49152` (`0xc000`) for `initcode`.
+
+## Abstract
+
+Enforce a maximum size limit (`max_initcode_size`) for `initcode`. If the size of `initcode` exceeds `max_initcode_size`, then contract creation fails with an out of gas error.
+
+Since [EIP-170](https://eips.ethereum.org/EIPS/eip-170) was implemented, there has been a size limit of `24576` (`0x6000`) on contract code. We propose to also limit the size of executable code to `2x` the above limit, i.e. `49152` (`0xc000`).
+
+This also leads to two nice properties:
+
+- instruction offset in code fits 16-bit value,
+- code size fits 16-bit value.
```

Comment:
Consider moving all of this to the motivation section.  Abstract should be a human-friendly and terse version of the specification, while the motivation section is where you state why this is a good idea, how these changes make things better, etc.